druid/src/widget/switch.rs

```diff
@@ -256,27 +275,33 @@ impl Widget<bool> for Switch {
         let is_active = ctx.is_active();
         let is_hovered = self.knob_hovered;
 
-        let normal_knob_gradient = LinearGradient::new(
-            UnitPoint::TOP,
-            UnitPoint::BOTTOM,
-            (
-                env.get(theme::FOREGROUND_LIGHT),
-                env.get(theme::FOREGROUND_DARK),
-            ),
-        );
-        let flipped_knob_gradient = LinearGradient::new(
-            UnitPoint::TOP,
-            UnitPoint::BOTTOM,
-            (
-                env.get(theme::FOREGROUND_DARK),
-                env.get(theme::FOREGROUND_LIGHT),
-            ),
-        );
-
-        let knob_gradient = if is_active {
-            flipped_knob_gradient
+        let knob_gradient = if ctx.is_disabled() {
```

Comment:
I think the most important thing here is that we not show the blue accent color, even if the switch is disabled in the 'on' state.